app/services/validator_check_active_service.rb

```diff
@@ -0,0 +1,85 @@
+class ValidatorCheckActiveService
+
+  def initialize
+    # If stake lower of eqal this value validator becomes inactive
+    @stake_exclude_height = 0
+
+    # if delinquent for this period of time validator becomes inactive
+    @delinquent_time = 24.hours
+  end
+
+  def update_validator_activity
+    Validator.all.each do |validator|
+      if validator.scorable?
+        check_if_should_be_scorable(validator)
```

Comment:
I would move all updates from `check_if_should_be_scorable` and split `check_if_should_be_scorable` to smaller checking methods, ie: 
```
validator.update(is_active: false) if smaller_check_method?
validator.update(is_rpc: true) if smaller_check_method_2?
```